lib/rom/repository/transaction.rb

```diff
@@ -0,0 +1,41 @@
+module ROM
+  class Repository
+    class Transaction
+      attr_reader :repo
+
+      attr_reader :ops
+
+      def initialize(repo, ops = [])
+        @repo = repo
+        @ops = ops
+      end
+
+      def create(changeset)
+        new(ops.dup << [:create, changeset])
```

Comment:
Could maybe do this as  `opts + [[:create, changeset]]` ... though, the double array there is too easy to miss, so I guess the extra dup call makes it more clear what's being avoided.

Still ... if each option returns a totally new command ... that more or less means that the contents of the transaction block _must_ be chained.  If you follow the rubyish sequence of commands, you'll drop the first one on the floor?